web2/hooks/useFlexiSwap.ts

```diff
@@ -0,0 +1,26 @@
+import { useMemo } from "react";
+import { useAuth } from "./useAuth";
+import { FlexiSwap } from "@/sdk/flexi-swap";
+import { FlexiSwapContract } from "@/sdk/contracts/flexi-swap-contract";
+import { FLEXISWAP_ADDRESS } from "@/sdk/constants";
+import { Approver } from "@/sdk/approver";
+
+const useFlexiSwap = () => {
+  const { isConnected } = useAuth();
+  const flexiSwapAddress: Address = useMemo(() => FLEXISWAP_ADDRESS, []);
+
+  const flexiSwap = useMemo(
+    () =>
+      isConnected
+        ? new FlexiSwap(
+            new FlexiSwapContract(flexiSwapAddress),
+            new Approver(flexiSwapAddress)
+          )
+        : null,
+    [isConnected, flexiSwapAddress]
+  );
```

Comment:
Lets do this without ternary

```suggestion
  const flexiSwap = useMemo(() => {
    if (!isConnected) return null;
    
    const flexiSwapContract = new FlexiSwapContract(flexiSwapAddress);
    const approver = new Approver(flexiSwapAddress);
    return new FlexiSwap(flexiSwapContract, approver);
  }, [isConnected, flexiSwapAddress]);
```